firebase/deployment/src/main/java/io/quarkiverse/googlecloudservices/firebase/deployment/FirebaseDevServiceConfig.java

```diff
@@ -0,0 +1,282 @@
+package io.quarkiverse.googlecloudservices.firebase.deployment;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+/**
+ * Root configuration class for Google Cloud Firebase that operates at build time.
+ * This class provides a nested structure for configuration, including
+ * a separate group for the development service configuration.
+ * <p>
+ * Here is an example of how to configure these properties:
+ * <p>
+ *
+ * <pre>
+ * quarkus.google.cloud.firebase.devservice.enabled = true
+ * quarkus.google.cloud.pubsub.devservice.image-name = gcr.io/google.com/cloudsdktool/google-cloud-cli # optional
+ * quarkus.google.cloud.pubsub.devservice.emulatorPort = 8085 # optional
+ * </pre>
+ */
+@ConfigMapping(prefix = "quarkus.google.cloud")
+@ConfigRoot
+public interface FirebaseDevServiceConfig {
+
+    /**
+     * Google Cloud project ID. The project is required to be set if you use the Firebase Auth Dev service.
+     */
+    Optional<String> projectId();
+
+    /**
+     * Configure the generic Firebase settings
+     */
+    Firebase firebase();
+
+    /**
+     * Configure the realtime database
+     */
+    Database database();
+
+    /**
+     * Configure the firestore
+     */
+    Firestore firestore();
+
+    /**
+     * Configure google cloud functions
+     */
+    Functions functions();
+
+    /**
+     * Configure Google Cloud Pub/Sub
+     */
+    PubSub pubsub();
+
+    /**
+     * Configure Google Cloud Storage
+     */
+    Storage storage();
+
+    interface Firebase {
+
+        /**
+         * Configuration for the firebase emulator devservice. This is the generic configuration for the firebase
+         * emulator. THe specifics are handled in each of the other dev services.
+         */
+        DevService devservice();
+
+        /**
+         * Configure the firebase auth settings
+         */
+        Auth auth();
+
+        /**
+         * Configure Firebase Hosting
+         */
+        Hosting hosting();
+
+        interface DevService {
+
+            /**
+             * Indicates to use the dev service for Firebase. The default value is true. This indicator is used
+             * to detect the Firebase DevService and disable the DevServices for extensions which conflict with the
+             * Firebase DevService.
+             */
+            @WithDefault("true")
+            boolean preferFirebaseDevServices();
+
+            /**
+             * Sets the Docker image name for the Google Cloud SDK.
+             * This image is used to emulate the Pub/Sub service in the development environment.
+             * The default value is 'gcr.io/google.com/cloudsdktool/google-cloud-cli'.
+             * <p>
+             * See also the documentation on Custom Docker images for more info about this image.
+             */
+            @WithDefault("node:23-alpine")
```

Comment:
The configured default didn't match the documentation default

Reply:
Good catch! Fixed